assets/js/settings/shared/utils.ts

```diff
@@ -25,6 +25,15 @@ export const getSetting = < T >(
 	return filter( value, fallback ) as T;
 };
 
+export const getSettingWithCoercion = < T >(
```

Comment:
Could we add some tests to this function? Similar to what we have for `getSetting()`:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/a180f22ecf0de60547cdc881bed1df6b4847a4a1/assets/js/settings/shared/test/get-setting.js